cpp/include/cudf/fixed_point/fixed_point.hpp

```diff
@@ -72,6 +72,371 @@ constexpr inline auto is_supported_representation_type()
+{
+  // See comments in divide_power10_128bit() for discussion.
+  // clang-format off
+  switch (exp10) {
+    case 0 : return value;
+    case 1 : return value * 10;
+    case 2 : return value * 100;
+    case 3 : return value * 1000;
+    case 4 : return value * 10000;
+    case 5 : return value * 100000;
+    case 6 : return value * 1000000;
+    case 7 : return value * 10000000;
+    case 8 : return value * 100000000;
+    case 9 : return value * 1000000000;
+    case 10: return value * 10000000000LL;
+    case 11: return value * 100000000000LL;
+    case 12: return value * 1000000000000LL;
+    case 13: return value * 10000000000000LL;
+    case 14: return value * 100000000000000LL;
+    case 15: return value * 1000000000000000LL;
+    case 16: return value * 10000000000000000LL;
+    case 17: return value * 100000000000000000LL;
+    case 18: return value * 1000000000000000000LL;
+    case 19: return value * large_power_of_10<19>();
+    case 20: return value * large_power_of_10<20>();
+    case 21: return value * large_power_of_10<21>();
+    case 22: return value * large_power_of_10<22>();
+    case 23: return value * large_power_of_10<23>();
+    case 24: return value * large_power_of_10<24>();
+    case 25: return value * large_power_of_10<25>();
+    case 26: return value * large_power_of_10<26>();
+    case 27: return value * large_power_of_10<27>();
+    case 28: return value * large_power_of_10<28>();
+    case 29: return value * large_power_of_10<29>();
+    case 30: return value * large_power_of_10<30>();
+    case 31: return value * large_power_of_10<31>();
+    case 32: return value * large_power_of_10<32>();
+    case 33: return value * large_power_of_10<33>();
+    case 34: return value * large_power_of_10<34>();
+    case 35: return value * large_power_of_10<35>();
+    case 36: return value * large_power_of_10<36>();
+    case 37: return value * large_power_of_10<37>();
+    case 38: return value * large_power_of_10<38>();
+    default: return 0;
+  }
+  // clang-format on
+}
+
+/**
+ * @brief Multiply an integer by a power of 10.
+ *
+ * @note Use this function if you have no a-priori knowledge of what exp10 might be.
+ * If you do, prefer calling the bit-size-specific versions
+ *
+ * @tparam Rep Representation type needed for integer exponentiation
+ * @tparam T Integral type of value to be multiplied.
+ * @param value The number to be multiplied.
+ * @param exp10 The power-of-10 of the multiplier.
+ * @return Returns value * 10^exp10
+ */
+template <typename Rep,
+          typename T,
+          typename cuda::std::enable_if_t<(cuda::std::is_integral_v<T>)>* = nullptr>
+CUDF_HOST_DEVICE inline constexpr T multiply_power10(T value, int exp10)
+{
+  // Use this function if you have no knowledge of what exp10 might be
+  // If you do, prefer calling the bit-size-specific versions
+  if constexpr (sizeof(Rep) <= 4) {
+    return multiply_power10_32bit(value, exp10);
+  } else if constexpr (sizeof(Rep) == 8) {
```

Comment:
Why is the above using `<= 4` but this is using `== 8`? Shouldn't both be `<=` or `==`?

Reply:
You could call it with a type where sizeof is 1 or 2, hence the <= 4.  I could change the other to <= 8, but I'm not anticipating us having any types with size in between.

Reply:
Let’s be consistent and use <=.